src/Instagram/index.jsx

```diff
@@ -0,0 +1,166 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React, { Component } from "react";
+// Runtime type checking for React props and similar objects
+import PropTypes from "prop-types";
+//> MDB
+// "Material Design for Bootstrap" is a great UI design framework
+import { MDBBtn, MDBIcon } from "mdbreact";
+
+//> PopupWindow
+// The window for handling oauth redirects
+import PopupWindow from "../PopupWindow";
+//> Utils
+import { toQuery, guidGenerator } from "../utils";
+//#endregion
+
+//#region > Components
+/**
+ * @class A button that is used for GitHub OAuth
+ */
+class OAuth2Login extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onBtnClick = this.onBtnClick.bind(this);
+    this.onRequest = this.onRequest.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onFailure = this.onFailure.bind(this);
+  }
+
+  onBtnClick() {
```

Comment:
When using `this.` I'd use arrow functions.

```suggestion
  onBtnClick = () => {
```